### DISCUSSION TITLE
How to access shared state from external process

### DISCUSSION THREAD

**Discussion by ptondereau** [tokio-rs/axum#1897] [Q&A]
I have a project with a Tauri app that starts an Axum server like this:
```rust
fn main() {
    tauri::Builder::default()
        .setup(|app| {
            let server = http_proxy::lauch(); // == bind().serve()
            tauri::async_runtime::spawn(async move {
                server.await.expect("failed to launch http proxy");
            });
            Ok(())
        })
        .invoke_handler(tauri::generate_handler![greet])
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}
```

`http-proxy` is a tiny HTTP proxy (catch-all) that just modify and adds some headers to intercepted requests.
In shared state, I would like to provide a solution that bypass this interception when a flag is false or true like this:
```rust
struct AppState {
  bypass: bool // default to false
}
```
and use it in my handler to early return untouched request. This way, it avoids IMHO start stop management of my proxy.

Do you know how to achieve this?
BTW, my first idea was to use channel, but I struggle a lot who is responsible for spawning tasks

**Comment by davidpdrsn** [comment]
Can you pass the state to `http_proxy::lauch`?

**Reply by ptondereau** [reply]
Yes I can but what about mutating it in tauri process? Is it safe toggle the `bypass`state parameter by just passing it?

**Reply by davidpdrsn** [reply]
I don't know. You might need an Arc<Mutex<_>>
